usl_pipeline/cloud_functions/main.py

```diff
@@ -136,6 +136,44 @@ def wrapper(cloud_event: functions_framework.CloudEvent) -> None:
     return decorator
 
 
+@functions_framework.cloud_event
+@_retry_and_report_errors()
+def build_and_upload_study_area_chunk(
+    cloud_event: functions_framework.CloudEvent,
+) -> None:
+    """Creates a study area chunk and uploads it to GCS.
+
+    This function is triggered when files containing raw geo data for a study area
+    are uploaded to GCS. It will group the files into TAR-files and upload it to a
+    bucket.
+    """
+    file_name = cloud_event.data["name"]
+    bucket_name = cloud_event.data["bucket"]
+
+    if re.search(file_names.WPS_DOMAIN3_NC_REGEX, file_name):
+        storage_client = storage.Client()
+        bucket = storage_client.bucket(bucket_name)
+        file_blob = bucket.blob(str(file_name))
+
+        with file_blob.open("rb") as input_file:
+            tar_fd = io.BytesIO()
+            with tarfile.open(mode="w", fileobj=tar_fd) as tar:
+                nc = tarfile.TarInfo(file_name)
+                nc.size = len(input_file)
+                tar.addfile(nc, io.BytesIO(input_file))
```

Comment:
You should be able to just pass `input_file` in there. I actually wouldn't expect wrapping it in a BytesIO object to work because the BytesIO constrcutor expects bytes, not a file
https://docs.python.org/3/library/io.html#io.BytesIO